src/control/server/mgmt_svc.go

```diff
@@ -197,6 +206,15 @@ func (svc *mgmtSvc) Join(ctx context.Context, req *mgmtpb.JoinReq) (*mgmtpb.Join
 		return nil, errors.Wrap(err, "unmarshal Join response")
 	}
 
+	// if join successful, record membership
+	if resp.GetStatus() == 0 && resp.GetState() == mgmtpb.JoinResp_IN {
```

Comment:
Would be nice to add a unit test for this case.

Reply:
yes will be adding test coverage when I had the capability to use the membership list

Reply:
OK with me.